### DISCUSSION TITLE
SPY/SPX OI to gex profile idea

### DISCUSSION THREAD

**Discussion by vannaonspeed** [2187Nick/thinkscript#8] [Ideas]
First of all, I wanna thank you for putting this out in the world, my theories and ideas for indicators on TOS far exceed my coding capabilities and I've tried for months to create the OI or volume as basically a volume profile to no avail so to stumble upon yours was a gift. Onto my idea; with what you've created for OI, in my mind it should be very simple to call upon the gamma at each strike, multiply by the OI (and multiply by 100, technically -100 for puts but not sure how well that would work with the put values going neg) in order to create the simple gex profile directly on the chart, however as it turns out, it isnt as simple as i had imagined and after a few hours of attempts, I thought I would refer back to you to see if it is indeed possible and how to go about it. I know you are working on a much larger project atm so if you do come up with something I would be happy to go through the code and add in those edits and share back here. If you dont end up having the time to look into it no sweat, and thank you for a wonderful tool or two. 
Cheers

**Comment by 2187Nick** [comment]
Hola.  This indicator adds gex on the chart. 
https://github.com/2187Nick/thinkscript/tree/main/gex_chart
![image](https://github.com/2187Nick/thinkscript/assets/75052782/99bd3267-f350-469c-a3de-d76ca7b1ad46)

**Reply by vannaonspeed** [reply]
Dunno how I glanced over this, you sir are a gentleman and a scholar, Cheers
